added conditional probability of n purchases for individual

MichaelSchreier opened the pull request.

CamDavidsonPilon commented on lifetimes/fitters/pareto_nbd_fitter.py:
```diff
@@ -282,3 +282,64 @@ def expected_number_of_purchases_up_to_time(self, t):
         second_term = 1 - (beta / (beta + t)) ** (s - 1)
         return first_term * second_term
+    
+        def conditional_probability_of_n_purchases_up_to_time(self, n, t, frequency, recency, T):
```
@MichaelSchreier, there is a syntax error here (bad indent)

CamDavidsonPilon commented on lifetimes/fitters/pareto_nbd_fitter.py:
```diff
@@ -231,9 +231,9 @@ def conditional_probability_alive(self, frequency, recency, T):
-        return 1. / (1. + (s / (r + s + x)) *
-                     (alpha + T) ** (r + x) * (beta + T) ** s * A_0)
+        A_0 = self._log_A_0([r, alpha, s, beta], x, t_x, T)
+        return 1. / (1. + exp(log(s) - log(r + s + x) +
```
👍 nice improvement

CamDavidsonPilon commented on lifetimes/fitters/pareto_nbd_fitter.py:
```diff
@@ -282,3 +282,64 @@ def expected_number_of_purchases_up_to_time(self, t):
+        zero = (n == 0) * (1 - exp(log_p_zero))
+        one = n * log(t) - log(np.math.factorial(n)) + log_B_one - log_l
+        two = log_B_two - log_l
+        three = misc.logsumexp([i * log(t) - log(np.math.factorial(i)) + log_B_three(i) - log_l for i in range(n+1)], axis=0)
```
can you pick different names here, like A, B, C, D? `one`, `two`, etc. carry too much addition meaning

CamDavidsonPilon commented:
Can you please add a test to `test_estimation.py`, ideally comparing against the paper's results or another implementation?

MichaelSchreier commented:
Jeez, both me and my git somehow screwed up here committing the files -.- Sorry for that!

Anyway, I fixed the indentation error (sloppy copy & paste job) and incorporated your other requests.
There are unfortunately no test cases in the source article so I went with testing the general properties that a probability should have. While writing the tests I also encountered some numerical stability issues which I then tried to fix as well.

The integration tests somehow fail on BetaGeoFitter which I did not touch here.

CamDavidsonPilon commented:
Closing this in favour of #155
